Lib/fontTools/subset/__init__.py

```diff
@@ -2915,32 +2901,10 @@ def prune_pre_subset(self, font, options):
 
 
 @_add_method(ttLib.getTableClass("name"))
-def prune_pre_subset(self, font, options):
-    nameIDs = set(options.name_IDs)
-    fvar = font.get("fvar")
-    if fvar:
-        nameIDs.update([axis.axisNameID for axis in fvar.axes])
-        nameIDs.update([inst.subfamilyNameID for inst in fvar.instances])
-        nameIDs.update(
-            [
-                inst.postscriptNameID
-                for inst in fvar.instances
-                if inst.postscriptNameID != 0xFFFF
-            ]
-        )
-    stat = font.get("STAT")
-    if stat:
-        if stat.table.AxisValueArray:
-            nameIDs.update(
-                [val_rec.ValueNameID for val_rec in stat.table.AxisValueArray.AxisValue]
-            )
-        nameIDs.update(
-            [axis_rec.AxisNameID for axis_rec in stat.table.DesignAxisRecord.Axis]
-        )
-    cpal = font.get("CPAL")
-    if cpal and cpal.version == 1:
-        nameIDs.update(cpal.paletteLabels)
-        nameIDs.update(cpal.paletteEntryLabels)
+def prune_post_subset(self, font, options):
+    visitor = NameRecordVisitor()
+    visitor.visit(font)
```

Comment:
why do we need to "visit" the entire font if we know already where we expect to find nameIDs?

Reply:
PTAL